packages/redux-dynostore-react-redux-subspace/src/subspaced.js

```diff
@@ -6,10 +6,26 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import isPlainObject from 'lodash.isplainobject'
 import { subspaced } from 'react-redux-subspace'
 
-const subspacedEnhancer = () => identifier => {
-  const subspaceEnhancer = subspaced(identifier)
+const subspacedEnhancer = ({ mapExtraState = () => null } = {}) => identifier => {
+  const mapState = (state, rootState) => {
+    const componentState = state[identifier]
+    if (!isPlainObject(componentState)) {
+      // perhaps a warning/error
```

Comment:
I think we should make a decision one way or the other on these warnings.  I'm thinking that this one is ok, as the slice state might legitimately not be a object, but the `extraState` should always return an object (otherwise how should it be merged?

```js
if (!isPlainObject(extraState)) {
  if (process.env.NODE_ENV !== 'production' && extraState !== null) {
    console.warn(`extra state must be a plain object but received ${extraState}`)
  }
  return componentState
}
```

I could be convinced that it should throw a `TypeError` instead of just warn.  @jpeyper?

Reply:
Let me know I'll gladly adapt the PR